pandas/tests/io/parser/test_na_values.py

```diff
@@ -421,3 +421,12 @@ def test_na_values_with_dtype_str_and_na_filter(all_parsers, na_filter):
 
     result = parser.read_csv(StringIO(data), na_filter=na_filter, dtype=str)
     tm.assert_frame_equal(result, expected)
+
+
+def test_cast_NA_to_bool_raises_error(c_parser_only):
+    parser = c_parser_only
```

Comment:
FYI: if you're using `c_parser_only`, move to `test_c_parser_only`.

Also, in light of my other comment [here](https://github.com/pandas-dev/pandas/pull/23968#issuecomment-442635986), let's try to get this test to pass with all engines.